test/x-content/build.gradle

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+apply plugin: 'elasticsearch.build'
+apply plugin: 'elasticsearch.publish'
+
+dependencies {
+  compileOnly project(":test:framework")
```

Comment:
Why is this `compileOnly`? In fact I don't think either of these should be `compileOnly` as there is no execution environment where either of these is implicitly "provided". They should both be `api` IMO.

Reply:
I moved them to `api` for now. I originally had them as compile only because I was planning to do shading and wanted to exclude them (since the test framework should already be provided). I didn't want the entire framework to get shaded. But that's for the future...